lib/wsdl.js

```diff
@@ -1422,18 +1423,23 @@ WSDL.prototype.xmlToObject = function(xml, callback) {
     var top = stack[stack.length - 1];
     var name = splitQName(top.schema).name,
       value;
-    if (name === 'int' || name === 'integer') {
-      value = parseInt(text, 10);
-    } else if (name === 'bool' || name === 'boolean') {
-      value = text.toLowerCase() === 'true' || text === '1';
-    } else if (name === 'dateTime' || name === 'date') {
-      value = new Date(text);
-    } else {
-      // handle string or other types
-      if (typeof top.object !== 'string') {
-        value = text;
+    if (self.options && self.options.customDeserializer && self.options.customDeserializer[ name ]) {
```

Comment:
no space between brackets for property access E.G. `self.options.customDeserializer[name]`